runtime/compiler/optimizer/HotFieldMarking.cpp

```diff
@@ -156,13 +150,21 @@ int32_t TR_HotFieldMarking::perform()
          {
          int32_t fieldOffset = (comp()->fej9()->getInstanceFieldOffset(itr->second->_clazz, itr->second->_fieldName, itr->second->_fieldNameLength, itr->second->_fieldSig, itr->second->_fieldSigLength) + TR::Compiler->om.objectHeaderSizeInBytes()) / TR::Compiler->om.sizeofReferenceField();
             
-         if (!comp()->fej9()->isAnonymousClass(itr->second->_clazz) && performTransformation(comp(), "%sUpdate hot field info for hot field. signature: %.*s; fieldName: %.*s; frequencyScore = %d\n", optDetailString(), itr->second->_fieldSigLength, itr->second->_fieldSig, itr->second->_fieldNameLength, itr->second->_fieldName, itr->second->_score) && (fieldOffset < U_8_MAX))
+         if (!comp()->fej9()->isAnonymousClass(itr->second->_clazz) && performTransformation(comp(), "%sUpdate hot field info for hot field. fieldSignature: %s; fieldName: %s; frequencyScore = %d\n", optDetailString(), itr->second->_fieldSig, itr->second->_fieldName, itr->second->_score) && (fieldOffset < U_8_MAX))
             {
             if (comp()->getOption(TR_TraceMarkingOfHotFields))
                {
                int32_t classNameLength = 0;
                char *className = comp()->fej9()->getClassNameChars(itr->second->_clazz, classNameLength);
-               traceMsg(comp(),"Hot field marked or updated. signature: %.*s; class:%.*s; fieldName: %.*s; frequencyScore = %d; fieldOffset: %d; \n", J9UTF8_LENGTH(itr->second->_fieldSig), J9UTF8_DATA(itr->second->_fieldSig), J9UTF8_LENGTH(className), J9UTF8_DATA(className), J9UTF8_LENGTH(itr->second->_fieldName), J9UTF8_DATA(itr->second->_fieldName), itr->second->_score, fieldOffset);
+               traceMsg(comp(), "<traceMarkingOfHotFields\n"
+                        "\tmethodSignature=\"%s\"\n"
+                        "\tmethodHotness=\"%s\">\n"
+                        "\tclassName=\"%s\">\n"
+                        "\tfieldName=\"%s\">"
+                        "\tfieldSig=\"%s\">"
+                        "\tfrequencyScore=%d>"
+                        "\tfieldOffset=%d>\n",
+                        comp()->signature(), comp()->getHotnessName(comp()->getMethodHotness()), className, itr->second->_fieldName, itr->second->_fieldSig, itr->second->_score, fieldOffset);
```

Comment:
This will produce output something like this:
```
<traceMarkingOfHotFields
	methodSignature="ONE"
	methodHotness="TWO">
	className="THREE">
	fieldName="FOUR">	fieldSig="FIVE">	frequencyScore=1000>	fieldOffset=-9999>
```
Is that what you're going for?  If so, do you mind if I ask what the significance of this formatting is?

Reply:
The main purpose is because having it all on the same line is hard to look at and pick out key information. The reasoning for having the last 4 items on the same line is to make it easier when debugging to grep for a specific field and offset combination. Do you have a suggestion that you would prefer to see? I got the idea to format the trace message like this from looking at other JIT trace messages that I thought were well-formatted in this same manner.

Reply:
Hm.  OK.  I thought it just looks a little odd with those unmatched `>` that begin with `methodHotness` and what their significance is.

Sorry, I'm not trying to be difficult here.

Reply:
Ah I see, don't worry it is no problem and is good attention to detail. I can certainly add the missing `>` to `methodSignature` as this was not intentionally missed.

Reply:
Would this be good now @0xdaryl?